modules/andes-core/broker/src/main/java/org/wso2/andes/server/handler/QueueDeclareHandler.java

```diff
@@ -58,137 +58,118 @@ public static QueueDeclareHandler getInstance()
 
     private final AtomicInteger _counter = new AtomicInteger();
 
-    public void methodReceived(AMQStateManager stateManager, QueueDeclareBody body, int channelId) throws AMQException
-    {
+    public void methodReceived(AMQStateManager stateManager, QueueDeclareBody body, int channelId) throws AMQException {
+
+        final AMQProtocolSession protocolConnection = stateManager.getProtocolSession();
+        final AMQSessionModel session = protocolConnection.getChannel(channelId);
+        VirtualHost virtualHost = protocolConnection.getVirtualHost();
+        ExchangeRegistry exchangeRegistry = virtualHost.getExchangeRegistry();
+        QueueRegistry queueRegistry = virtualHost.getQueueRegistry();
+        DurableConfigurationStore store = virtualHost.getDurableConfigurationStore();
+
+        final AMQShortString queueName;
+
+        // if we aren't given a queue name, we create one which we return to the client
+        if ((body.getQueue() == null) || (body.getQueue().length() == 0)) {
```

Comment:
Shall we flip the body.getQueue().length() == 0 to  0 == body.getQueue().length()